packages/rOpenScPCA/tests/testthat/test-evaluate-clusters.R

```diff
@@ -0,0 +1,38 @@
+suppressPackageStartupMessages(library(SingleCellExperiment))
+
+set.seed(2024)
+sce <- splatter::simpleSimulate(nGenes = 1000, verbose = FALSE) |>
+  scater::logNormCounts() |>
+  scater::runPCA(ncomponents = 10)
+test_mat <- reducedDim(sce, "PCA")
+
+
+cluster_df <- calculate_clusters(test_mat)
+cluster_df$random_extra_column <- 10
+
+
+test_that("calculate_silhouette works as expected", {
+  df <- calculate_silhouette(test_mat, cluster_df)
+
+  expect_setequal(
+    colnames(df),
+    c("cell_id", "cluster", "algorithm", "weighting", "nn", "resolution", "silhouette_width", "random_extra_column")
+  )
+  expect_equal(df$cell_id, rownames(test_mat))
+  expect_equal(df$cluster, cluster_df$cluster)
+  expect_type(df$silhouette_width, "double") # TODO is expect_s3_class with "numeric" safer here? i imagine no decimals is near impossible?
```

Comment:
Ah, it's not an S3 object so that doesn't work. So I went with [`expect_vector()`](https://testthat.r-lib.org/reference/expect_vector.html) to test if its `"numeric"`.